helix-term/src/commands/lsp.rs

```diff
@@ -1324,3 +1331,191 @@ pub fn select_references_to_symbol_under_cursor(cx: &mut Context) {
         },
     );
 }
+
+pub fn compute_inlay_hints_for_all_view(editor: &mut Editor, jobs: &mut crate::job::Jobs) {
+    if !editor.config().lsp.display_inlay_hints {
+        for (view, _) in editor.tree.views_mut() {
+            view.last_inlay_hint_request_id = None;
+            for doc in editor.documents.values_mut() {
+                doc.reset_all_text_annotations();
+            }
+        }
+        return;
+    }
+
+    for (view, _) in editor.tree.views_mut() {
+        let doc = match editor.documents.get_mut(&view.doc) {
+            Some(doc) => doc,
+            None => continue,
+        };
+        if let Some(callback) = compute_inlay_hints_for_view(view, doc) {
+            jobs.callback(callback);
+        }
+    }
+}
+
+fn compute_inlay_hints_for_view(
+    view: &mut View,
+    doc: &mut Document,
+) -> Option<std::pin::Pin<Box<impl Future<Output = Result<crate::job::Callback, anyhow::Error>>>>> {
+    let view_id = view.id;
+    let doc_id = view.doc;
+    let doc_revision = doc.get_current_revision();
+
+    let language_server = doc.language_server()?;
+
+    let capabilities = language_server.capabilities();
+
+    let (future, new_inlay_hint_request) = match capabilities.inlay_hint_provider {
+        Some(
+            lsp::OneOf::Left(true)
+            | lsp::OneOf::Right(lsp::InlayHintServerCapabilities::Options(_)),
+        ) => {
+            let doc_text = &doc.text();
+            let len_lines = doc_text.len_lines();
+
+            // Compute ~3 times the current view height of inlay hints, that way some scrolling
+            // will not show half the view with hints and half without while still being faster
+            // than computing all the hints for the full file (which could be dozens of time
+            // longer than the view is).
+            let view_height = view.inner_height();
+            let first_visible_line = view.estimate_last_doc_line(doc).saturating_sub(view_height);
```

Comment:
This is a bit backwards as `estimate_last_doc_line` interally calcualtes the first line and then adds `view_height`.
Instead just use:
```suggestion
            let first_visible_line = doc.text().char_to_line(view.offfset.anchor);
```

Reply:
Fixed